src/System.Security.Cryptography.Pkcs/tests/CertLoader.cs

```diff
@@ -0,0 +1,150 @@
+                CheckIfCertWasLeftInCertStore();
+                _alreadySearchedMyStore = true;
+            }
+
+            CertLoadMode testMode = CertLoader.TestMode;
+            switch (testMode)
+            {
+                case CertLoadMode.Disable:
+                    return null;
+
+                case CertLoadMode.LoadFromPfx:
+                    return new X509Certificate2(PfxData, Password);
+
+                case CertLoadMode.LoadFromStore:
+                    {
+                        X509Certificate2Collection matches = FindMatches(CertLoader.StoreName, StoreLocation.CurrentUser);
+                        if (matches.Count == 1)
+                            return matches[0];
+
+                        if (matches.Count == 0)
+                            throw new Exception($"No matching certificate found in store {CertLoader.StoreName}");
+                        else
+                            throw new Exception($"Multiple matching certificates found in store {CertLoader.StoreName}");
+                    }
+
+                default:
+                    throw new Exception($"Unexpected CertLoader.TestMode value: {testMode}");
+            }
+        }
+
+        public abstract byte[] CerData { get; }
+        public abstract byte[] PfxData { get; }
+        public abstract string Password { get; }
+
+        private void CheckIfCertWasLeftInCertStore()
+        {
+            X509Certificate2Collection matches = FindMatches("MY", StoreLocation.CurrentUser);
+            if (matches.Count == 0)
+            {
+                matches = FindMatches("MY", StoreLocation.LocalMachine);
+            }
+
+            if (matches.Count != 0)
+            {
+                X509Certificate2 cer = new X509Certificate2(CerData);
+                string issuer = cer.Issuer;
+                string serial = cer.SerialNumber;
+
+                throw new Exception($"A certificate issued to {issuer} with serial number {serial} was found in your Personal certificate store. This will cause some tests to fail due to machine configuration. Please remove these.");
+            }
+        }
+
+        private X509Certificate2Collection FindMatches(string storeName, StoreLocation storeLocation)
+        {
+            using (X509Certificate2 cer = new X509Certificate2(CerData))
+            {
+                string issuer = cer.Issuer;
+                string serial = cer.SerialNumber;
+
+                using (X509Store store = new X509Store(storeName, storeLocation))
+                {
+                    store.Open(OpenFlags.ReadOnly | OpenFlags.OpenExistingOnly);
+
+                    X509Certificate2Collection matches = store.Certificates.Find(X509FindType.FindByIssuerDistinguishedName, issuer, validOnly: false);
+                    matches = matches.Find(X509FindType.FindBySerialNumber, serial, validOnly: false);
```

Comment:
Why not just use foreach (var storeCert in store.Certificates) { if (storeCert.Equals(cer)) { ... } } ?